Allow user to specify additional forms in their civicrm.settings.php

webmaster-cses-org-uk opened the pull request:
This proposed change introduces a mechanism for users to specify their own additional form class names in their civicrm.settings.php file.

The use case that motivates this change is a third-party extension (in this case, Grant Application Forms) that creates its own form types that incorporate profiles, and may include an organisation / current_employer field.

Tested and working on CiviCRM 5.37.0.

webmaster-cses-org-uk committed:
Allow user to specify additional forms in their civicrm.settings.php

Add a setting 'additionalForms' to the settings array that allows a user to specify their own additional forms with which to use this feature.

Useful if third-party extensions create their own forms that accept profiles (for example, Grant Application Pages).

Could extend to allow more advanced features e.g. wildcards / regex / match all or exclude forms

webmaster-cses-org-uk committed:
Update publicautocomplete.php

webmaster-cses-org-uk committed:
Update documentation for user-specified additional forms

Accompanies proposed changes to publicautocomplete.php

twomice commented:
Question for @tttp : Can you help me understand why we're limiting to certain forms in the first place? That is, if something went wrong and this feature began trying to work on some un-tested form, what is the downside? Is it just that we haven't tested on those forms so behavior could be unpredictable? Or is it conceivably something more harmful?

@webmaster-cses-org-uk  Seems good in principle. Thanks for the contribution.  You'll see I have a question out to another maintainer (above). Once I've got that sorted, I may ask that you correct this PR for compliance with CiviCRM's Coding Standards. (See specifically the section on tools which can help make that easier: https://docs.civicrm.org/dev/en/latest/standards/#tools)

webmaster-cses-org-uk committed:
Update for CiviCRM coding standards

Update proposed additions to _publicautocomplete_supported_forms() to comply with CiviCRM coding standards (specifically, use of line breaks in control structures).

webmaster-cses-org-uk commented:
@twomice Thank you for the update. I have pre-emptively updated for coding standards, to save time later! Thanks.

twomice commented:
Thanks for that update @webmaster-cses-org-uk Not hearing any objections from @tttp I'm merging this.
